File: app/src/main/java/com/hillal/acc/ui/accounts/EditAccountFragment.java
package com.hillal.acc.ui.accounts;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.SwitchCompat;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;
import androidx.core.view.ViewCompat;

import com.hillal.acc.R;
import com.hillal.acc.data.model.Account;
import com.hillal.acc.viewmodel.AccountViewModel;
import com.hillal.acc.databinding.FragmentAddAccountBinding;

public class EditAccountFragment extends Fragment {
    private FragmentAddAccountBinding binding;
    private AccountViewModel accountViewModel;
    private long accountId;
    private Account oldAccount;
    private SwitchCompat whatsappSwitch;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        accountViewModel = new ViewModelProvider(this).get(AccountViewModel.class);
        if (getArguments() != null) {
            accountId = getArguments().getLong("accountId", -1);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentAddAccountBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setupViews();
        loadAccount();
        // ضبط insets للجذر لرفع المحتوى مع الكيبورد وأزرار النظام
        ViewCompat.setOnApplyWindowInsetsListener(view, (v, insets) -> {
            int bottom = insets.getInsets(androidx.core.view.WindowInsetsCompat.Type.ime()).bottom;
            if (bottom == 0) {
                bottom = insets.getInsets(androidx.core.view.WindowInsetsCompat.Type.systemBars()).bottom;
            }
            v.setPadding(v.getPaddingLeft(), v.getPaddingTop(), v.getPaddingRight(), bottom);
            return insets;
        });
    }

    private void setupViews() {
        whatsappSwitch = binding.whatsappSwitch;
        
        // تعيين النص بشكل صريح
        whatsappSwitch.setText(getString(R.string.whatsapp_enabled));
        
        binding.saveButton.setOnClickListener(v -> updateAccount());
        binding.cancelButton.setOnClickListener(v -> requireActivity().onBackPressed());
    }

    private void loadAccount() {
        if (accountId != -1) {
            accountViewModel.getAccountById(accountId).observe(getViewLifecycleOwner(), account -> {
                if (account != null) {
                    oldAccount = account;
                    binding.nameEditText.setText(account.getName());
                    binding.phoneEditText.setText(account.getPhoneNumber());
                    binding.openingBalanceEditText.setText(String.valueOf(account.getBalance()));
                    binding.notesEditText.setText(account.getNotes());
                    whatsappSwitch.setChecked(account.isWhatsappEnabled());
                }
            });
        }
    }

    private void updateAccount() {
        String name = binding.nameEditText.getText().toString();
        String phone = binding.phoneEditText.getText().toString();
        String notes = binding.notesEditText.getText().toString();
        String balanceStr = binding.openingBalanceEditText.getText().toString();
        boolean whatsappEnabled = whatsappSwitch.isChecked();

        if (name.isEmpty()) {
            binding.nameEditText.setError("الرجاء إدخال اسم الحساب");
            return;
        }

        try {
            double balance = Double.parseDouble(balanceStr);
            accountViewModel.getAccountById(accountId).observe(getViewLifecycleOwner(), account -> {
                if (account != null) {
                    if (oldAccount == null) {
                        Toast.makeText(getContext(), "حدث خطأ في تحميل بيانات الحساب الأصلية", Toast.LENGTH_SHORT).show();
                        return;
                    }
                    account.setName(name);
                    account.setPhoneNumber(phone);
                    account.setBalance(balance);
                    account.setNotes(notes);
                    account.setWhatsappEnabled(whatsappEnabled);
                    account.setUpdatedAt(System.currentTimeMillis());
                    account.setServerId(oldAccount.getServerId()); 
                    account.setSyncStatus(0);
                    accountViewModel.updateAccount(account);
                    Toast.makeText(getContext(), R.string.account_saved, Toast.LENGTH_SHORT).show();
                    Navigation.findNavController(requireView()).navigateUp();
                }
            });
        } catch (NumberFormatException e) {
            binding.openingBalanceEditText.setError("الرجاء إدخال رصيد صحيح");
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
} 

File: app/src/main/java/com/hillal/acc/ui/accounts/EditAccountFragment.kt
package com.hillal.acc.ui.accounts

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.view.ViewCompat
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.findNavController
import com.hillal.acc.R
import com.hillal.acc.data.model.Account
import com.hillal.acc.databinding.FragmentAddAccountBinding
import com.hillal.acc.viewmodel.AccountViewModel

class EditAccountFragment : Fragment() {
    private var _binding: FragmentAddAccountBinding? = null
    private val binding get() = _binding!!
    private lateinit var accountViewModel: AccountViewModel
    private var accountId: Long = -1
    private var oldAccount: Account? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        accountViewModel = ViewModelProvider(this)[AccountViewModel::class.java]
        arguments?.let {
            accountId = it.getLong("accountId", -1)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentAddAccountBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupViews()
        loadAccount()
        // ضبط insets للجذر لرفع المحتوى مع الكيبورد وأزرار النظام
        ViewCompat.setOnApplyWindowInsetsListener(view) { v, insets ->
            var bottom = insets.getInsets(androidx.core.view.WindowInsetsCompat.Type.ime()).bottom
            if (bottom == 0) {
                bottom = insets.getInsets(androidx.core.view.WindowInsetsCompat.Type.systemBars()).bottom
            }
            v.setPadding(v.paddingLeft, v.paddingTop, v.paddingRight, bottom)
            insets
        }
    }

    private fun setupViews() {
        binding.whatsappSwitch.text = getString(R.string.whatsapp_enabled)
        binding.saveButton.setOnClickListener { updateAccount() }
        binding.cancelButton.setOnClickListener { requireActivity().onBackPressedDispatcher.onBackPressed() }
    }

    private fun loadAccount() {
        if (accountId != -1L) {
            accountViewModel.getAccountById(accountId).observe(viewLifecycleOwner) { account ->
                account?.let {
                    oldAccount = it
                    binding.nameEditText.setText(it.name)
                    binding.phoneEditText.setText(it.phoneNumber)
                    binding.openingBalanceEditText.setText(it.balance.toString())
                    binding.notesEditText.setText(it.notes)
                    binding.whatsappSwitch.isChecked = it.isWhatsappEnabled
                }
            }
        }
    }

    private fun updateAccount() {
        val name = binding.nameEditText.text.toString()
        val phone = binding.phoneEditText.text.toString()
        val notes = binding.notesEditText.text.toString()
        val balanceStr = binding.openingBalanceEditText.text.toString()
        val whatsappEnabled = binding.whatsappSwitch.isChecked

        if (name.isEmpty()) {
            binding.nameEditText.error = "الرجاء إدخال اسم الحساب"
            return
        }

        val balance = balanceStr.toDoubleOrNull()
        if (balance == null) {
            binding.openingBalanceEditText.error = "الرجاء إدخال رصيد صحيح"
            return
        }

        accountViewModel.getAccountById(accountId).observe(viewLifecycleOwner) { account ->
            if (account != null) {
                if (oldAccount == null) {
                    Toast.makeText(context, "حدث خطأ في تحميل بيانات الحساب الأصلية", Toast.LENGTH_SHORT).show()
                    return@observe
                }
                account.name = name
                account.phoneNumber = phone
                account.balance = balance
                account.notes = notes
                account.isWhatsappEnabled = whatsappEnabled
                account.updatedAt = System.currentTimeMillis()
                account.serverId = oldAccount!!.serverId
                account.syncStatus = 0
                accountViewModel.updateAccount(account)
                Toast.makeText(context, getString(R.string.account_saved), Toast.LENGTH_SHORT).show()
                findNavController().navigateUp()
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
} 